tests/std/tests/P0645R10_text_formatting_legacy_text_encoding/test.cpp

```diff
@@ -8,55 +8,48 @@
 
 #include "test_format_support.hpp"
 
+#pragma warning(disable : 5105) // macro expansion producing 'defined' has undefined behavior
+#define CODEPAGE(x) (static_cast<unsigned int>(x) ? static_cast<unsigned int>(x) : 932u)
+#include "../../../../stl/src/format.cpp"

```

Comment:
I'm 99% certain this dirty testing hack won't work internally; needs a rethink.

There's an `activeCodePage` element for application manifests, but I've seen suggestions that it can only be set to `utf-8`. That would be fine for the `formatting_utf8_text_encoding` test below, but not for this one. I'll have to investigate more fully now that I've realized this cross-inclusion trick won't work.

Reply:
I'd have the override in format.h guarded by _STL_INTERNAL_TESTING

Reply:
Although this is the sort of thing that could quite quickly become a "feature"

Reply:
Having `<format>` detect a macro that the tests define is unusual, but not utterly unprecedented (we have `_STL_INTERNAL_CHECK` and `_STL_INTERNAL_STATIC_ASSERT`, controlled by `_ENABLE_STL_INTERNAL_CHECK`). If we don't document it, we don't need to worry about supporting it. I would prefer such a "test hook" macro over this weird hack.

(Note that `MSVC_INTERNAL_TESTING` doesn't appear in product code and should be regarded as a temporary workaround whenever it appears.)

Reply:
yeah, my only other thought was some kind of extremely cursed linker magic to override the code-page function. I think it's fine if we have limited test coverage of the actual importlib stub, it doesn't really change based on the active code page. The manifest entry is only utf-8, and changing the system code page requires a reboot.